helpers/controllers/passwordController.ts

```diff
@@ -0,0 +1,108 @@
+import db from '../dbload'
+import _ from 'lodash'
+import bcrypt from 'bcrypt'
+import { nanoid } from 'nanoid'
+import { UserInputError, AuthenticationError } from 'apollo-server-micro'
+import { changeChatPassword } from '../mattermost'
+import { Context } from '../../@types/helpers'
+import { sendResetEmail } from '../mail'
+import { decode, encode } from '../encoding'
+import { passwordValidation } from '../formValidation'
+
+const { User } = db
+const THREE_DAYS = 1000 * 60 * 60 * 24 * 3
+
+export const reqPwReset = async (
+  _parent: void,
+  arg: { userOrEmail: string },
+  ctx: Context
+) => {
+  const { req } = ctx
+  try {
+    const userOrEmail = _.get(arg, 'userOrEmail', null)
+    if (!userOrEmail) {
+      throw new UserInputError('Please provide username or email')
+    }
+
+    let user: any
+    if (userOrEmail.indexOf('@') !== -1) {
```

Comment:
this could be a really useful util function, like `const user = getUserByEmailOrUsername( userOrEmail )`

Util function could do the throwing and input validation, making this `reqPwReset` function much smaller.

Reply:
If you make this a util function, we could potentially use this in login as well!